src/legacy/core_plugins/vis_default_editor/public/components/controls/raw_json.tsx

```diff
@@ -34,48 +32,77 @@ function RawJsonParamEditor({
   setValue,
   setTouched,
 }: AggParamEditorProps<string>) {
-  const label = (
-    <>
-      <FormattedMessage id="visDefaultEditor.controls.jsonInputLabel" defaultMessage="JSON input" />{' '}
-      <EuiIconTip
-        position="right"
-        content={i18n.translate('visDefaultEditor.controls.jsonInputTooltip', {
-          defaultMessage:
-            "Any JSON formatted properties you add here will be merged with the elasticsearch aggregation definition for this section. For example 'shard_size' on a terms aggregation.",
-        })}
-        type="questionInCircle"
-      />
-    </>
+  const [isFieldValid, setFieldValidity] = useState(true);
+  const [editorReady, setEditorReady] = useState(false);
+
+  const editorTooltipText = useMemo(
+    () =>
+      i18n.translate('visDefaultEditor.controls.jsonInputTooltip', {
+        defaultMessage:
+          "Any JSON formatted properties you add here will be merged with the elasticsearch aggregation definition for this section. For example 'shard_size' on a terms aggregation.",
+      }),
+    []
   );
-  const isValid = isValidJson(value);
 
-  const onChange = (ev: React.ChangeEvent<HTMLTextAreaElement>) => {
-    const textValue = ev.target.value;
-    setValue(textValue);
-    setValidity(isValidJson(textValue));
-  };
+  const jsonEditorLabelText = useMemo(
+    () =>
+      i18n.translate('visDefaultEditor.controls.jsonInputLabel', {
+        defaultMessage: 'JSON input',
+      }),
+    []
+  );
 
-  useEffect(() => {
-    setValidity(isValid);
-  }, [isValid]);
+  const label = useMemo(
+    () => (
+      <>
+        {jsonEditorLabelText}{' '}
+        <EuiIconTip position="right" content={editorTooltipText} type="questionInCircle" />
+      </>
+    ),
+    []
```

Comment:
Since every value referenced inside the function should appear in the dependencies array, let's add them:
```suggestion
    [jsonEditorLabelText, editorTooltipText]
```


Moreover when the `react-hooks/exhaustive-deps` rule is enabled in the [config](https://github.com/elastic/kibana/blob/master/.eslintrc.js#L84), eslint will warn due to missing dependencies. So let's prevent it.